libs/datamodel/core/src/lib.rs

```diff
@@ -1,3 +1,70 @@
+//! The data structures in this crate are organized into 3 layers:
+//! * The AST layer contains the data data structures representing the schema input.
+//! * The model layer contains the data structures that are semantically rich and therefore engines can build upon.
+//! * The JSON layer contains the data structures that represent the contract for the DMMF.
+//!
+//! The responsibilities of each top level module is as following:
+//! * `common`: contains constants and generic helpers
+//! * `error`: contains the error and result types
+//! * `ast`: contains the data structures for the AST of a Prisma schema. And the parsing functions to turn an input string into an AST.
+//! * `dml`: contains the models representing the Datamodel part of a Prisma schema
+//! * `configuration`: contains the models representing the Datasources and Generators of a Prisma schema
+//! * `transform`: contains the logic to turn an AST into models and vice versa
+//! * `json`: contains the logic to turn models into their JSON/DMMF representation
+//!
+//! The flow between the layers is depicted in the following diagram.
+//!
+//!                ┌──────────────────┐
```

Comment:
How is this rendered in rustdoc? Should this be captured in a code block of type `txt`?

Reply:
I don't know. Is this important for internal docs? I thought we would look at them directly in the source. Or is there a benefit to publish them to `docs.rs` and look at them there?

Reply:
> Or is there a benefit to publish them to docs.rs and look at them there?

Oh, I was thinking about the way this would render in `cargo doc` locally.

Reply:
ah never used this locally. Will give that a try.

Reply:
Looks beautiful imo :-D 

<img width="988" alt="image" src="https://user-images.githubusercontent.com/2470881/88925597-02e83380-d275-11ea-90a2-a5a05bb10d0d.png">